ownCloud/Client/PhotoAlbumTableViewController.swift

```diff
@@ -0,0 +1,195 @@
+//
+//  PhotoAlbumViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 27.02.2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import Photos
+
+extension PHAssetCollection {
+	var assetCount: Int {
+		// Some collections can't return the estimated count, therefore we have to fetch the actual assets to get the count
+		if self.estimatedAssetCount != NSNotFound {
+			return self.estimatedAssetCount
+		} else {
+			let options = PHFetchOptions()
+			options.predicate = NSPredicate(format:  "mediaType = %d || mediaType = %d", PHAssetMediaType.image.rawValue, PHAssetMediaType.video.rawValue)
+			return PHAsset.fetchAssets(in: self, options: options).count
+		}
+	}
+
+	func fetchThumbnailAsset() -> PHAsset? {
+		// Fetch only one / most recent asset which shall represent the collection as a thumbnail
+		let fetchOptions = PHFetchOptions()
+		fetchOptions.sortDescriptors = [NSSortDescriptor(key: "creationDate", ascending: false)]
+		fetchOptions.fetchLimit = 1
+		let fetchResult = PHAsset.fetchAssets(in: self, options: fetchOptions)
+		return fetchResult.firstObject
+	}
+}
+
+class PhotoAlbumTableViewController : UITableViewController, Themeable {
+
+	class PhotoAlbum {
+		var name: String?
+		var count: Int?
+		var thumbnail: UIImage?
+		var collection: PHAssetCollection?
+
+		var countString : String {
+			if count != nil {
+				return "\(count!)"
+			} else {
+				return ""
+			}
+		}
+	}
+
+	var albums = [PhotoAlbum]()
+
+	var fetchAlbumQueue = DispatchQueue(label: "com.owncloud.photoalbum.queue", qos: DispatchQoS.userInitiated)
+
+	// The queue which is used to fetch thumbnails uses lowest priority, to ensure that scrolling is smooth
+	var fetchAlbumThumbnailsQueue = DispatchQueue(label: "com.owncloud.photoalbum.queue", qos: DispatchQoS.background)
+
+	// This is used just to pass the selection callback to PhotoSelectionViewController when an album is selected
+	var selectionCallback :PhotosSelectedCallback?
+
+	// MARK: - UIViewController lifecycle
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		self.title = "Albums".localized
+		self.tableView.rowHeight = PhotoAlbumTableViewCell.cellHeight
+		self.tableView.register(PhotoAlbumTableViewCell.self, forCellReuseIdentifier: PhotoAlbumTableViewCell.identifier)
+		self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancel))
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+		Theme.shared.register(client: self)
```

Comment:
Not a problem right now, but when adding the option to preview items in the future, `Theme.shared.register(client: self)` and `Theme.shared.unregister(client: self)` might get called multiple times as the `PhotoAlbumTableViewController` disappears and reappears. While that should work, it'll call `applyThemeCollection` without necessity.

Moving `Theme.shared.register(client: self)` to `viewDidLoad` and `Theme.shared.unregister(client: self)` to `deinit` should work better in those scenarios. 

Partly off-topic: I just checked and saw other classes registering and unregistering in `viewWillAppear` and `viewWillDisappear` as well. Might be worth addressing this in a small, separate PR to minimize theme/redraw overhead there as well. (/cc @hosy)

Reply:
Addressed this in issue #298

Reply:
@felix-schwarz fixed